### ISSUE TITLE
ATV will not connect to local Library

### ISSUE THREAD

**Issue by DawgBiscuit** [iBaa/PlexConnect#423]
So it took me a long time to finally get PlexConnect functioning with Icon HiJack etc... OpenPlex shows the status as running and things seem fine on the first ATV I setup.  I have Plex running successfully on the ATV using standard iMovie Hijack.  I can see/watch Shared Libraries from my friends, I can play Music off my local Plex Server, but when I click on Discover it just sits there forever.

Mac Mini is address: 192.168.10.5 (Static)
ATV is 192.168.10.35 (Dynamic)
DNS is running and the ATV is using that successfully 
Webserver at same address is running fine
All my normal Plex Clients on iOS and OS X connect perfectly

But for the ATV my Movie Library does not show up.  Only the remote shares as stated above.

I am running the Plex Server and OpenPlex on the same Mac Mini.  I just cannot get the ATV to pull the movies.

I am a Networking IT professional so totally get all the networking side of things.  I redid the settings.cfg to point to the local IP address that is in the same subnet.  I also verified that I can login into Plex Server using that local IP address rather than just localhost.

I ran a Packet Capture and see the ATV trying to connect to the Mac Mini but nothing ever pulls up.  Other than that the ATV functions totally normal. And again it runs all the other things off Plex including Music off the same server as my Movies.

Logs:
00:07:16 PlexConnect: started: 00:07:16
00:07:16 PlexConnect: Version: 0.5-ThemeDev-260116
00:07:16 PlexConnect: Python: 2.7.9 (v2.7.9:648dcafa7e5f, Dec 10 2014, 10:10:46) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
00:07:16 PlexConnect: Host OS: darwin
00:07:16 PlexConnect: Fanart Support: PIL installed? True
00:07:16 PlexConnect: IP_self (from settings): 192.168.10.5
00:07:16 DNSServer: started: 00:07:16
00:07:16 DNSServer: ***
00:07:16 DNSServer: DNSServer: Serving DNS on 192.168.10.5 port 53.
00:07:16 DNSServer: ***
00:07:16 WebServer: started: 00:07:16
00:07:16 WebServer: ***
00:07:16 WebServer: WebServer: Serving HTTP on 192.168.10.5 port 80.
00:07:16 WebServer: ***
00:07:16 WebServer: started: 00:07:16
00:07:16 WebServer: ***
00:07:16 WebServer: WebServer: Serving HTTPS on 192.168.10.5 port 443.
00:07:16 WebServer: ***
00:07:24 PlexAPI: ***
00:07:24 PlexAPI: poke plex.tv - request Plex Media Server list
00:07:24 PlexAPI: ***
00:07:25 PlexAPI: Servers (local, plex.tv, MyPlex): 3
00:39:09 PlexAPI: ***
00:39:09 PlexAPI: poke plex.tv - request Plex Media Server list
00:39:09 PlexAPI: ***
00:39:10 PlexAPI: Servers (local, plex.tv, MyPlex): 3
00:40:33 PlexAPI: ***
00:40:33 PlexAPI: poke plex.tv - request Plex Media Server list
00:40:33 PlexAPI: ***
00:40:34 PlexAPI: Servers (local, plex.tv, MyPlex): 3
00:41:18 PlexAPI: ***
00:41:18 PlexAPI: poke plex.tv - request Plex Media Server list
00:41:18 PlexAPI: ***
00:41:20 PlexAPI: Servers (local, plex.tv, MyPlex): 3
00:44:32 PlexAPI: ***
00:44:32 PlexAPI: poke plex.tv - request Plex Media Server list
00:44:32 PlexAPI: ***
00:44:33 PlexAPI: Servers (local, plex.tv, MyPlex): 3

Settings.cfg:
[PlexConnect]
enable_plexgdm = True
ip_pms = 192.168.10.5
port_pms = 32400
enable_dnsserver = True
port_dnsserver = 53
ip_dnsmaster = 8.8.8.8
prevent_atv_update = True
intercept_atv_icon = True
icon = com.apple.imovietheatre.appletv
enable_plexconnect_autodetect = False
ip_plexconnect = 192.168.10.5
hosttointercept = www.icloud.com
port_webserver = 80
enable_webserver_ssl = True
port_ssl = 443
certfile = ./assets/certificates/trailers.pem
allow_gzip_atv = False
allow_gzip_pmslocal = False
allow_gzip_pmsremote = True
loglevel = Normal
logpath = .

Short amount of Packet Capture:
--- Start Of Stream ---
23:41:12.437801 Acknowledgment RA:b8:78:2e:39:c7:5e 
23:41:12.454356 Acknowledgment RA:b8:78:2e:39:c7:5e 
23:41:13.878317 Action (02:18:5a:59:8f:f0): Spectrum Management Act#2
23:41:14.819432 CF +QoS ARP, Request who-has 192.168.10.35 tell 10.128.128.128, length 46
23:41:16.915749 Action (02:18:5a:59:8f:f0): Spectrum Management Act#2
23:41:17.964510 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [S.], seq 1953265520, ack 280551509, win 65535, options [mss 1460,nop,wscale 5,nop,nop,TS val 309526374 ecr 815451687,sackOK,eol], length 0
23:41:17.964552 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [S.], seq 1953265520, ack 280551509, win 65535, options [mss 1460,nop,wscale 5,nop,nop,TS val 309526498 ecr 815451687,sackOK,eol], length 0
23:41:18.000972 Acknowledgment RA:b8:78:2e:39:c7:5e 
23:41:17.971052 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 1, win 4117, options [nop,nop,TS val 309526590 ecr 815451904], length 0
23:41:17.979189 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 218, win 4110, options [nop,nop,TS val 309526598 ecr 815451911], length 0
23:41:18.022462 Clear-To-Send RA:b8:78:2e:39:c7:5e 
23:41:17.979673 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [P.], seq 1:971, ack 218, win 4110, options [nop,nop,TS val 309526598 ecr 815451911], length 970
23:41:17.999358 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 485, win 4102, options [nop,nop,TS val 309526617 ecr 815451931], length 0
23:41:17.999471 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 491, win 4102, options [nop,nop,TS val 309526617 ecr 815451931], length 0
23:41:17.999678 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 544, win 4100, options [nop,nop,TS val 309526617 ecr 815451931], length 0
23:41:18.005563 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [P.], seq 971:1030, ack 544, win 4100, options [nop,nop,TS val 309526623 ecr 815451931], length 59
23:41:18.013336 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], ack 917, win 4089, options [nop,nop,TS val 309526630 ecr 815451943], length 0
23:41:18.035009 Clear-To-Send RA:b8:78:2e:39:c7:5e 
23:41:18.014485 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [P.], seq 1030:1120, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 90
23:41:18.015530 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 1120:2568, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.015563 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 2568:4016, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.016329 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 4016:5464, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.016365 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 5464:6912, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.016386 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 6912:8360, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.016409 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [.], seq 8360:9808, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 1448
23:41:18.016424 CF +QoS IP 192.168.10.5.443 > 192.168.10.35.49783: Flags [P.], seq 9808:9868, ack 917, win 4096, options [nop,nop,TS val 309526631 ecr 815451943], length 60
23:41:18.054702 Acknowledgment RA:b8:78:2e:39:c7:5e 
23:41:18.074153 Acknowledgment RA:b8:78:2e:39:c7:5e 
23:41:18.074828 Clear-To-Send RA:b8:78:2e:39:c7:5e

**Comment by iBaa** [comment]
Couple of notes:
- ip_pms/port_pms is not used, if PlexGDM is used: enable_plexgdm
- PlexGDM is no more used, if you sign in to MyPlex - which you seemingly have, otherwise you wouldn't have "shared servers".
- the PMS addresses are provided by MyPlex (https://plex.tv/api/resources?includeHttps=1) - which may include "external" sub-addresses of host plex.direct for encryption purposes. Your router has to be aware of these and allow their usage - topic "DNS rebind protection".
Ah... and please post your questions in the plex forums - bigger audience, quicker answer, I guess.

**Comment by DawgBiscuit** [comment]
So not sure exactly what the outcome should be…

But I I have "enable_plexgdm = False" now

I set the ip_pms back to the original setting of 192.168.178.10.  I assume that is the default

I still had to set ip_plexconnect = 192.168.10.5 (my mac mini) or the OpenPlex would not start.

I am still able to see channels etc…  but when I click on Discover nothing seems to come up.  Any other suggestions?

 |Terry Morgan | 601.454.7262 | Email: terrymorgan@me.com

**Comment by DawgBiscuit** [comment]
Actually I set the enableplexconnectauotdetect = True (I had it False)
And ip+plexconnect = 0.0.0.0 

The console logs show things seem to be working right.  I can see that when I change options new logs appear.

But I am still not seeing content.


 |Terry Morgan | 601.454.7262 | Email: terrymorgan@me.com

**Comment by iBaa** [comment]
Let me rephrase my notes:
- At the beginning, PMS was identified by a fix address: Setting `ip_pms`:`port_ms`
- Lateron PlexGDM came along. If enabled (`enable_plexgdm`), it overrides the ip:port setting, as it discovers your PMS automagically.
- Then MyPlex was born. If signed in, PlexConnect retrieves your PMS setup from plex.tv, address shown above. It overrides both ip:port and PlexGDM settings, effectively rendering them unused.

These settings (plus the MyPlex sign in part) are for PlexConnect connecting to your PMSs (owned and shared ones).

Settings `ip_plexconnect`, `enable_plexconnect_autodetect` are different. Those are used to help PlexConnect to identify itself - or the host machine it is running on. Typically it can figure it out on its own... but in some cases it can't (like with multiple network connections, it might pick the wrong interface), and needs your help picking the correct address (enable=False; ip='your selected address').

In fact, if you get an response on your aTV like the PlexConnect menu bar plus "PMS: No items available" the later part around the PlexConnect IP should be fine...

But still, the connection to your local PMS is broken (only yours, remote/shared servers are fine), right? Again, my bet is on "DNS rebind protection" (there should be a white list option in your router), but I really can't be sure. We need a better log (`loglevel=High`) of the discovery process.
